src/DataModel/IMetadataProvider.ts

```diff
@@ -0,0 +1,30 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import "whatwg-fetch";
+import {IData} from "./IData";
+
+/**
+ * @interface Describes an abstract meta-data providing facility which translates from raw @link Response to an abstract data model.
+ */
+export interface IMetadataProvider
```

Comment:
Well - framing alone won't do it. There is already a very simple attempt to that topic in `JsonLdMetadataProvider.ts` (i.e. `parseMetadata`)